### DISCUSSION TITLE
[Help Wanted] No expand/collapse button in workspace viewer

### DISCUSSION THREAD

**Discussion by TTTPOB** [REditorSupport/vscode-R#1077] [Q&A] [answered]
## my situation
![image](https://user-images.githubusercontent.com/19909103/163512310-62eb6224-a419-442c-a4be-d41f8bc4ac34.png)

## normal situation
![image](https://user-images.githubusercontent.com/19909103/163512413-cb422435-a8a2-4a40-a031-353bf38b19ea.png)

**Comment by ElianHugh** [comment, marked as answer]
What is your level of detail setting? It looks like it's on minimal rather than detailed. See: `r.session.levelOfObjectDetail`

**Reply by TTTPOB** [reply]
Thank you for the quick answer! I don't know there is a setting controls this before.
It solved my problem, thank you.

**Reply by ElianHugh** [reply]
No worries! If you ever have any questions feel free to ask or check out the wiki! All the best

**Comment by danielbasso** [comment]
I've been thinking for some time to suggest that the default option for `vsc.str.max.level` should be 2. I think it's way more probable that a person would like it to be fully expandable then not. @ElianHugh , what do you think?

**Reply by renkun-ken** [reply]
`vsc.str.max.level=2` has some significant performance issue if user loads a data of many columns, a list of many elements, or nested levels in the R session. Currently, we write all column information to `{tempDir}/workspace.json` on each top-level user input via a task callback. We had that for some while but revert to `0` for lower delay of user input.

**Reply by ElianHugh** [reply]
Right, yeah, it was a concious decision due to significant performance issues for some users whenever the callback occurred.  IIRC, this was chosen around the same time as the object limit was implemented

**Reply by lukesdm** [reply]
It seems calling `options(vsc.str.max.level = 2)` in the current version of VS Code with this extension breaks the Workspace viewer.

Example: trying to expand 'my_variables'.

Before:  
![image](https://user-images.githubusercontent.com/2824165/169713623-b4438f96-1e4b-40d9-ad65-63370b1eedfb.png)

After:  
![image](https://user-images.githubusercontent.com/2824165/169713641-0d404237-f237-410f-b1d0-9988c876168d.png)


Environment:
* VS Code 1.67.2  
* REditorSupport 2.5.0
* Windows 10


Edit: I see similar behavior with changing `r.session.levelOfObjectDetail` to non minimal. It could be related to my data - I'm dealing with geospatial shp files.
